src/System.Private.ServiceModel/tools/setupfiles/sync-pr.cmd

```diff
@@ -0,0 +1,43 @@
+@echo off 
+setlocal
+
+REM 
+REM This calls a web service, specified on the command line, which syncs a 
+REM remote server's repo to the correct PR. 
+REM This script is meant to be called from a GitHub Pull Request. 
+REM If you must, you can fake it by setting the 'ghprbPullId' environment variable
+REM 
+
+set __args=%*
+
+echo.
+echo  WCF Remote Github Repo sync script
+
+if '%__args%'=='' (
+    echo   [%~n0] A URL must be specified for the pull request synchronization URL
+    echo     Usage:  %~n0 [sync url]
+    echo   sync-url - URL on remote server for PR synchronization
+    echo.
+    echo   Example:  %~n0 http://wcfci-sync-server/repo/sync.ashx
+    set __EXITCODE=1
+    goto done
+)
+
+set __SYNC_HOST_URL=%1
+
+if '%ghprbPullId%'=='' (
+    echo   [%~n0] This script should only be called only from the context of a GitHub Pull Request
```

Comment:
I'm more of a fan of the "self contained" approach personally. I've had way too many times where I was trying to locally repro something and found the scripts unnavigable due to magic. 

In general, I think an optional parameter helps us in the case where we need to specify a clear set of repro steps running against a server. 

For day to day use, I will allow the ghprb param to be set automatically.